modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/SourceModelParser.java

```diff
@@ -150,9 +150,10 @@ public interface SourceModelParser extends SemanticTermsParser, StereotypeModelP
   SourceClusterSupportModelProperty getSourceClusterSupportModelProperty();
 
   /**
-   * @return a {@link MuleVersion} representing the minimum mule version this component can run on
+   * @return a {@link MinMuleVersionResult} that contains the minimum mule version this component can run on and the reason why
+   *         that version was assigned.
```

Comment:
Shouldn't this still be Optional<MinMuleVersionResult> ? Why is the possibility of empty value eliminated.

Reply:
We discussed it above: it always returned a value, so it didn't make sense to make it an optional.

Reply:
Will ParameterParsers go through the same change? I see in this PR that they are unchanged. 

In my opinion, the parameter parser should at least return an Optional<MinMuleVersionResult>. What do you think?

Reply:
I think I would either remove the methods or make them return the default min mule version. Seems inconsistent to make them Optional while in others they aren't optional.